awx/ui_next/src/screens/Template/shared/SurveyList.jsx

```diff
@@ -1,48 +1,259 @@
+    }, [template.id]),
+    { questions: [], name: '', description: '' }
   );
 
   useEffect(() => {
     fetchSurvey();
   }, [fetchSurvey]);
-  if (contentError) {
-    return <ContentError error={contentError} />;
-  }
-  if (isLoading) {
-    return <ContentLoading />;
-  }
 
-  return (
-    questions?.length > 0 &&
-    questions.map((question, index) => (
+  const isAllSelected =
+    selected.length === questions?.length && selected.length > 0;
+
+  const {
+    isLoading: isDeleteLoading,
+    deleteItems: deleteQuestions,
+    deletionError,
+    clearDeletionError,
+  } = useDeleteItems(
+    useCallback(async () => {
+      if (isAllSelected) {
+        return JobTemplatesAPI.destroySurvey(template.id);
+      }
+      const surveyQuestions = [];
+      questions.forEach(q =>
+        !selected.find(s => s.id === q.id) ? surveyQuestions.push(q) : null
+      );
+      return JobTemplatesAPI.updateSurvey(template.id, {
+        name,
+        description,
+        spec: surveyQuestions,
+      });
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [selected]),
+    {
+      qsConfig: QS_CONFIG,
+      fetchItems: fetchSurvey,
+    }
+  );
+  const {
+    isToggleLoading,
+    error: toggleError,
+    request: toggleSurvey,
+  } = useRequest(
+    useCallback(async () => {
+      await JobTemplatesAPI.update(template.id, {
+        survey_enabled: !surveyEnabled,
+      });
+      return setSurveyEnabled(!surveyEnabled);
+    }, [template, surveyEnabled]),
+    template.survey_enabled
+  );
+
+  useEffect(() => {
+    if (toggleError) {
+      setShowToggleError(true);
+    }
+  }, [toggleError]);
+
+  const handleSelectAll = isSelected => {
+    setSelected(isSelected ? [...questions] : []);
+  };
+
+  const handleSelect = item => {
+    if (selected.some(s => s.id === item.id)) {
+      setSelected(selected.filter(s => s.id !== item.id));
+    } else {
+      setSelected(selected.concat(item));
+    }
+  };
+
+  const handleDelete = async () => {
+    await deleteQuestions();
+    setIsDeleteModalOpen(false);
+    setSelected([]);
+  };
+  let content;
+  if (
+    (isLoading || isToggleLoading || isDeleteLoading) &&
+    questions?.length <= 0
+  ) {
+    content = <ContentLoading />;
+  } else if (contentError) {
+    content = <ContentError error={contentError} />;
+  } else if (!questions || questions?.length <= 0) {
+    content = (
+      <ContentEmpty
+        title={i18n._(t`No Survey Questions Found`)}
+        message={i18n._(t`Please add survey questions.`)}
+      />
+    );
+  } else {
+    content = questions?.map((question, index) => (
       <SurveyListItem
         key={question.id}
         isLast={index === questions.length - 1}
         isFirst={index === 0}
         question={question}
+        isChecked={selected.some(s => s.id === question.id)}
+        onSelect={() => handleSelect(question)}
       />
-    ))
+    ));
+  }
+  return (
+    <>
+      <DataToolbar id="survey-toolbar">
```

Comment:
This should be its own component.

Reply:
Yeah I was thinking that too, but was trying to limit the scope of this PR.  However, with that said I'm also not sure that adding a new component expands the scope, but it does expand the number of files.

Reply:
consider it it's own component